cts-java/src/test/java/org/ga4gh/cts/api/references/ReferenceSetsPagingIT.java

```diff
@@ -0,0 +1,132 @@
+     */
+    @Test
+    public void checkPagingOneByOneThroughReferenceSets() throws AvroRemoteException {
+
+        // retrieve them all
+        final List<ReferenceSet> listOfReferenceSets = Utils.getAllReferenceSets(client);
+        assertThat(listOfReferenceSets).isNotEmpty();
+
+        // we will remove ReferenceSets from this Set and assert at the end that we have zero
+        final Set<ReferenceSet> setOfReferenceSets = new HashSet<>(listOfReferenceSets);
+        assertThat(listOfReferenceSets).hasSize(setOfReferenceSets.size());
+
+        // page through the ReferenceSets using the same query parameters
+        String pageToken = null;
+        for (ReferenceSet ignored : listOfReferenceSets) {
+            final SearchReferenceSetsRequest pageReq =
+                    SearchReferenceSetsRequest.newBuilder()
+                                           .setPageSize(1)
+                                           .setPageToken(pageToken)
+                                           .build();
+            final SearchReferenceSetsResponse pageResp = client.references.searchReferenceSets(pageReq);
+            final List<ReferenceSet> pageOfReferenceSets = pageResp.getReferenceSets();
+            pageToken = pageResp.getNextPageToken();
+
+            assertThat(pageOfReferenceSets).hasSize(1);
+            assertThat(setOfReferenceSets).contains(pageOfReferenceSets.get(0));
+
+            setOfReferenceSets.remove(pageOfReferenceSets.get(0));
+        }
+
+        assertThat(pageToken).isNull();
+        assertThat(setOfReferenceSets).isEmpty();
+    }
+
+    /**
+     * Check that we can page through the {@link ReferenceSet}s we receive from
+     * {@link Client.References#searchReferenceSets(SearchReferenceSetsRequest)}
+     * using an increment as large as the non-paged set of results.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void checkPagingByOneChunkThroughReferenceSets() throws AvroRemoteException {
+
+        final List<ReferenceSet> listOfReferenceSets = Utils.getAllReferenceSets(client);
+        assertThat(listOfReferenceSets).isNotEmpty();
+
+        // page through the reference sets in one gulp
+        checkSinglePageOfReferenceSets(listOfReferenceSets.size(),
+                                       listOfReferenceSets);
+    }
+
+    /**
+     * Check that we can page through the {@link ReferenceSet}s we receive from
+     * {@link Client.References#searchReferenceSets(SearchReferenceSetsRequest)}
+     * using an increment twice as large as the non-paged set of results.
+     *
+     * @throws AvroRemoteException if there's a communication problem or server exception ({@link GAException})
+     */
+    @Test
+    public void checkPagingByOneTooLargeChunkThroughReferenceSets() throws AvroRemoteException {
```

Comment:
This is similar. All we can assert here is that we should get back all the results after paging. The sizes of the pages are up to the server, and anything > 0 is a valid value.